java.lang.NullPointerException - spring boot + gradle plugin

### Search first

- [X] I searched and no similar issues were found

### Description

I'm trying to integrate the Gradle plugin into my dev stack, but I facing a weird exception that I didn't find online.

Stack:
- Spring Boot 3.1.4
- Hibernate, PostgreSQL
- Kotlin
- Gradle written in Kotlin DSL

I tried two ways for configuration
1. liquidbase.properties
2. custom task in gradle

Both ways produce the same error, so I think it's a bug probably.


liquibase.properties
```properties
classpath=src/main/resources
changeLogFile=src/main/resources/db/migrations.yaml
url=jdbc:postgresql://localhost:5432/my-database
username=postgres
password=root
driver=org.postgresql.Driver
referenceUrl=hibernate:spring:com.graet?dialect=org.hibernate.dialect.PostgreSQL95Dialect&hibernate.physical_naming_strategy=org.springframework.boot.orm.jpa.hibernate.SpringPhysicalNamingStrategy&hibernate.implicit_naming_strategy=org.springframework.boot.orm.jpa.hibernate.SpringImplicitNamingStrategy
referenceDriver=liquibase.ext.hibernate.database.connection.HibernateDriver
defaultSchemaName=public
referenceLiquibaseSchemaName=tools
```

When I run: `./gradlew diffChangelog` there is the issue

Terminal log:
```
landsman@M2 backend % ./gradlew diffChangelog
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :diffChangelog FAILED
liquibase-plugin: Running the 'main' activity...
####################################################
##   _     _             _ _                      ##
##  | |   (_)           (_) |                     ##
##  | |    _  __ _ _   _ _| |__   __ _ ___  ___   ##
##  | |   | |/ _` | | | | | '_ \ / _` / __|/ _ \  ##
##  | |___| | (_| | |_| | | |_) | (_| \__ \  __/  ##
##  \_____/_|\__, |\__,_|_|_.__/ \__,_|___/\___|  ##
##              | |                               ##
##              |_|                               ##
##                                                ##
##  Get documentation at docs.liquibase.com       ##
##  Get certified courses at learn.liquibase.com  ##
##  Free schema change activity reports at        ##
##      https://hub.liquibase.com                 ##
##                                                ##
####################################################
Starting Liquibase at 11:37:20 (version 4.20.0 #7837 built at 2023-03-07 16:25+0000)
Liquibase Version: 4.20.0
WARNING: License service not loaded, cannot determine Liquibase Pro license status. Please consider re-installing Liquibase to include all dependencies. Continuing operation without Pro license.
[2023-10-16 11:37:20] INFO [liquibase.integration] Starting command execution.
[2023-10-16 11:37:20] WARNING [liquibase.configuration] Potentially ignored key(s) in file exists at path liquibase.properties
 - 'referenceLiquibaseSchemaName'
[2023-10-16 11:37:20] INFO [liquibase.servicelocator] Cannot load service: liquibase.database.Database: liquibase.ext.hibernate.database.HibernateSpringBeanDatabase Unable to get public no-arg constructor
[2023-10-16 11:37:20] INFO [liquibase.servicelocator] Cannot load service: liquibase.database.Database: liquibase.ext.hibernate.database.HibernateSpringPackageDatabase Unable to get public no-arg constructor
[2023-10-16 11:37:20] WARNING [liquibase.database] Unknown database: Hibernate
[2023-10-16 11:37:20] INFO [liquibase.database] Error getting default schema
java.lang.NullPointerException: Cannot invoke "java.sql.CallableStatement.executeQuery()" because "cs" is null
        at liquibase.executor.jvm.JdbcExecutor$QueryCallableStatementCallback.doInCallableStatement(JdbcExecutor.java:546)
        at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:111)
        at liquibase.executor.jvm.JdbcExecutor.query(JdbcExecutor.java:198)
        at liquibase.executor.jvm.JdbcExecutor.query(JdbcExecutor.java:209)
        at liquibase.executor.jvm.JdbcExecutor.queryForObject(JdbcExecutor.java:217)
        at liquibase.executor.jvm.JdbcExecutor.queryForObject(JdbcExecutor.java:232)
        at liquibase.executor.jvm.JdbcExecutor.queryForObject(JdbcExecutor.java:227)
        at liquibase.database.AbstractJdbcDatabase.getConnectionSchemaName(AbstractJdbcDatabase.java:374)
        at liquibase.database.AbstractJdbcDatabase.getDefaultSchemaName(AbstractJdbcDatabase.java:331)
        at liquibase.CatalogAndSchema.customize(CatalogAndSchema.java:153)
        at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:181)
        at liquibase.command.core.DiffCommandStep.generateDatabaseShapshot(DiffCommandStep.java:191)
        at liquibase.command.core.DiffCommandStep.createReferenceSnapshot(DiffCommandStep.java:178)
        at liquibase.command.core.DiffCommandStep.createDiffResult(DiffCommandStep.java:101)
        at liquibase.command.core.DiffCommandStep.run(DiffCommandStep.java:86)
        at liquibase.command.CommandScope.execute(CommandScope.java:195)
        at liquibase.integration.commandline.CommandRunner.call(CommandRunner.java:55)
        at liquibase.integration.commandline.CommandRunner.call(CommandRunner.java:24)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
        at picocli.CommandLine.execute(CommandLine.java:2170)
        at liquibase.integration.commandline.LiquibaseCommandLine.lambda$null$2(LiquibaseCommandLine.java:367)
        at liquibase.Scope.child(Scope.java:196)
        at liquibase.Scope.child(Scope.java:172)
        at liquibase.integration.commandline.LiquibaseCommandLine.lambda$execute$3(LiquibaseCommandLine.java:332)
        at liquibase.Scope.child(Scope.java:196)
        at liquibase.Scope.child(Scope.java:172)
        at liquibase.integration.commandline.LiquibaseCommandLine.execute(LiquibaseCommandLine.java:330)
        at liquibase.integration.commandline.LiquibaseCommandLine.main(LiquibaseCommandLine.java:91)

[2023-10-16 11:37:20] SEVERE [liquibase.integration] Cannot invoke "java.sql.ResultSet.next()" because "catalogs" is null
liquibase.exception.CommandExecutionException: java.lang.NullPointerException: Cannot invoke "java.sql.ResultSet.next()" because "catalogs" is null
        at liquibase.command.CommandScope.execute(CommandScope.java:209)
        at liquibase.integration.commandline.CommandRunner.call(CommandRunner.java:55)
        at liquibase.integration.commandline.CommandRunner.call(CommandRunner.java:24)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
        at picocli.CommandLine.execute(CommandLine.java:2170)
        at liquibase.integration.commandline.LiquibaseCommandLine.lambda$null$2(LiquibaseCommandLine.java:367)
        at liquibase.Scope.child(Scope.java:196)
        at liquibase.Scope.child(Scope.java:172)
        at liquibase.integration.commandline.LiquibaseCommandLine.lambda$execute$3(LiquibaseCommandLine.java:332)
        at liquibase.Scope.child(Scope.java:196)
        at liquibase.Scope.child(Scope.java:172)
        at liquibase.integration.commandline.LiquibaseCommandLine.execute(LiquibaseCommandLine.java:330)
        at liquibase.integration.commandline.LiquibaseCommandLine.main(LiquibaseCommandLine.java:91)
Caused by: java.lang.NullPointerException: Cannot invoke "java.sql.ResultSet.next()" because "catalogs" is null
        at liquibase.snapshot.jvm.JdbcSnapshotGenerator.getDatabaseCatalogNames(JdbcSnapshotGenerator.java:146)
        at liquibase.snapshot.jvm.CatalogSnapshotGenerator.snapshotObject(CatalogSnapshotGenerator.java:35)
        at liquibase.snapshot.jvm.JdbcSnapshotGenerator.snapshot(JdbcSnapshotGenerator.java:65)
        at liquibase.snapshot.SnapshotGeneratorChain.snapshot(SnapshotGeneratorChain.java:49)
        at liquibase.snapshot.DatabaseSnapshot.include(DatabaseSnapshot.java:313)
        at liquibase.snapshot.DatabaseSnapshot.init(DatabaseSnapshot.java:100)
        at liquibase.snapshot.DatabaseSnapshot.<init>(DatabaseSnapshot.java:59)
        at liquibase.snapshot.JdbcDatabaseSnapshot.<init>(JdbcDatabaseSnapshot.java:34)
        at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:214)
        at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:187)
        at liquibase.command.core.DiffCommandStep.generateDatabaseShapshot(DiffCommandStep.java:191)
        at liquibase.command.core.DiffCommandStep.createReferenceSnapshot(DiffCommandStep.java:178)
        at liquibase.command.core.DiffCommandStep.createDiffResult(DiffCommandStep.java:101)
        at liquibase.command.core.DiffCommandStep.run(DiffCommandStep.java:86)
        at liquibase.command.CommandScope.execute(CommandScope.java:195)
        ... 18 more


Unexpected error running Liquibase: Cannot invoke "java.sql.ResultSet.next()" because "catalogs" is null

liquibase.exception.CommandExecutionException: java.lang.NullPointerException: Cannot invoke "java.sql.ResultSet.next()" because "catalogs" is null
        at liquibase.command.CommandScope.execute(CommandScope.java:209)
        at liquibase.integration.commandline.CommandRunner.call(CommandRunner.java:55)
        at liquibase.integration.commandline.CommandRunner.call(CommandRunner.java:24)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
        at picocli.CommandLine.execute(CommandLine.java:2170)
        at liquibase.integration.commandline.LiquibaseCommandLine.lambda$null$2(LiquibaseCommandLine.java:367)
        at liquibase.Scope.child(Scope.java:196)
        at liquibase.Scope.child(Scope.java:172)
        at liquibase.integration.commandline.LiquibaseCommandLine.lambda$execute$3(LiquibaseCommandLine.java:332)
        at liquibase.Scope.child(Scope.java:196)
        at liquibase.Scope.child(Scope.java:172)
        at liquibase.integration.commandline.LiquibaseCommandLine.execute(LiquibaseCommandLine.java:330)
        at liquibase.integration.commandline.LiquibaseCommandLine.main(LiquibaseCommandLine.java:91)
Caused by: java.lang.NullPointerException: Cannot invoke "java.sql.ResultSet.next()" because "catalogs" is null
        at liquibase.snapshot.jvm.JdbcSnapshotGenerator.getDatabaseCatalogNames(JdbcSnapshotGenerator.java:146)
        at liquibase.snapshot.jvm.CatalogSnapshotGenerator.snapshotObject(CatalogSnapshotGenerator.java:35)
        at liquibase.snapshot.jvm.JdbcSnapshotGenerator.snapshot(JdbcSnapshotGenerator.java:65)
        at liquibase.snapshot.SnapshotGeneratorChain.snapshot(SnapshotGeneratorChain.java:49)
        at liquibase.snapshot.DatabaseSnapshot.include(DatabaseSnapshot.java:313)
        at liquibase.snapshot.DatabaseSnapshot.init(DatabaseSnapshot.java:100)
        at liquibase.snapshot.DatabaseSnapshot.<init>(DatabaseSnapshot.java:59)
        at liquibase.snapshot.JdbcDatabaseSnapshot.<init>(JdbcDatabaseSnapshot.java:34)
        at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:214)
        at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:187)
        at liquibase.command.core.DiffCommandStep.generateDatabaseShapshot(DiffCommandStep.java:191)
        at liquibase.command.core.DiffCommandStep.createReferenceSnapshot(DiffCommandStep.java:178)
        at liquibase.command.core.DiffCommandStep.createDiffResult(DiffCommandStep.java:101)
        at liquibase.command.core.DiffCommandStep.run(DiffCommandStep.java:86)
        at liquibase.command.CommandScope.execute(CommandScope.java:195)
        ... 18 more

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':diffChangelog'.
> Process 'command '/opt/homebrew/Cellar/openjdk/20.0.1/libexec/openjdk.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 7s
4 actionable tasks: 1 executed, 3 up-to-date
```

Tasks are registered in Gradle correctly:

```
landsman@M2 backend % ./gradlew tasks

> Task :tasks

------------------------------------------------------------
Tasks runnable from root project 'api'
------------------------------------------------------------

Application tasks
-----------------
bootRun - Runs this project as a Spring Boot application.
bootTestRun - Runs this project as a Spring Boot application using the test runtime classpath.

Build tasks
-----------
assemble - Assembles the outputs of this project.
bootBuildImage - Builds an OCI image of the application using the output of the bootJar task
bootJar - Assembles an executable jar archive containing the main classes and their dependencies.
build - Assembles and tests this project.
buildDependents - Assembles and tests this project and all projects that depend on it.
buildKotlinToolingMetadata - Build metadata json file containing information about the used Kotlin tooling
buildNeeded - Assembles and tests this project and all projects it depends on.
classes - Assembles main classes.
clean - Deletes the build directory.
jar - Assembles a jar archive containing the main classes.
kotlinSourcesJar - Assembles a jar archive containing the sources of target 'kotlin'.
resolveMainClassName - Resolves the name of the application's main class.
resolveTestMainClassName - Resolves the name of the application's test main class.
testClasses - Assembles test classes.

Build Setup tasks
-----------------
init - Initializes a new Gradle build.
wrapper - Generates Gradle wrapper files.

Documentation tasks
-------------------
javadoc - Generates Javadoc API documentation for the main source code.

Help tasks
----------
buildEnvironment - Displays all buildscript dependencies declared in root project 'api'.
dependencies - Displays all dependencies declared in root project 'api'.
dependencyInsight - Displays the insight into a specific dependency in root project 'api'.
dependencyManagement - Displays the dependency management declared in root project 'api'.
help - Displays a help message.
javaToolchains - Displays the detected java toolchains.
kotlinDslAccessorsReport - Prints the Kotlin code for accessing the currently available project extensions and conventions.
outgoingVariants - Displays the outgoing variants of root project 'api'.
projects - Displays the sub-projects of root project 'api'.
properties - Displays the properties of root project 'api'.
resolvableConfigurations - Displays the configurations that can be resolved in root project 'api'.
tasks - Displays the tasks runnable from root project 'api'.

Liquibase tasks
---------------
calculateChecksum - Calculate and print a checksum for the <liquibaseCommandValue> changeset.
changelogSync - Mark all changes as executed in the database.
changelogSyncSql - Output the raw SQL Liquibase would use when running changelogSync to the <liquibaseOutputFile> or STDOUT if no file is specified.
changelogSyncToTag - Mark all undeployed changesets up to and including the <liquibaseCommandValue> tag as executed in your database.
changelogSyncToTagSql - Output the raw SQL used by Liquibase when running changelogSyncToTag to <liquibaseOutputFile> or STDOUT if no file is specified.
checks - Execute the <liquibaseCommandValue> quality check
clearChecksums - Remove all saved checksums from the database. On next run checksums will be recomputed.  Useful for "MD5Sum Check Failed" errors.
dbDoc - Generates Javadoc-like documentation for the existing database and changelogs to the <liquibaseCommandValue> directory.
deactivateChangelog - Removes the changelogID from your changelog so it stops sending reports to Liquibase Hub.
diff - Compare two databases and write differences to <liquibaseOutputFile> or STDOUT of no file is specified.
diffChangelog - Compare two databases to produce changesets and write them to a changelog file
dropAll - Drop all database objects owned by the user. Note that functions, procedures and packages are not dropped (Liquibase limitation)
executeSql - Execute a SQL string given in <liquibaseCommandValue>"
executeSqlFile - Execute a SQL file given in <liquibaseCommandValue>"
futureRollbackCountSql - Generates SQL to sequentially revert <liquibaseCommandValue> changes the database.
futureRollbackFromTagSql - Generates SQL to revert future changes up to the <liquibaseCommandValue> tag, and writes it to <liquibaseOutputFile>, or STDOUT if no file is specified.
futureRollbackSql - Generate the raw SQL needed to rollback undeployed changes to <liquibaseOutputFile> or to STDOUT if no value is specified.
generateChangelog - Generate a Groovv changelog from the current state of the database to <LiquibaseOutputFile> or STDOUT of no file is specified.
history - List all deployed changesets and their deployment IDs to <liquibaseOutputFile> or STDOUT if no file is provided.
listLocks - List the hostname, IP address, and timestamp of the Liquibase lock record.
markNextChangesetRan - Mark the next change you apply as executed in the database.
markNextChangesetRanSql - Write SQL to mark the next change you apply as executed in the database to <liquibaseOutputFile>, or STDOUT if no value is provided.
registerChangelog - Register the changelog with a Liquibase Hub project.
releaseLocks - Remove the Liquibase lock record from the DATABASECHANGELOG table.
rollback - Rollback changes made to the database since the the <liquibaseCommandValue> tag was applied.
rollbackCount - Rollback the last <liquibaseCommandValue> changes.
rollbackCountSql - Write SQL to roll back the last <liquibaseCommandValue> changes to <liquibaseOutputFile> or STDOUT if no file is specified.
rollbackOneChangeset - Roll back a specific changeset, without rolling back changesets deployed before or afterwards. (Liquibase Pro key required)
rollbackOneChangesetSql - Write SQL to roll back a specific changeset to <liquibaseCommandValue> in the <LiquibaseOutputFile> file (or STDOUT if not specified), without rolling back changesets deployed before or afterwards. (Liquibase Pro key required)
rollbackOneUpdate - Rollback one update from the database (Liquibase Pro key required).
rollbackOneUpdateSql - Write SQL to rollback one update from the database to <LiquibaseOutputFile or STDOUT if not specified (Liquibase Pro key required).
rollbackSql - Generate SQL to roll back the database to the state it was in when the <liquibaseCommandValue> tag was applied.
rollbackToDate - Rollback changes made to the database since the <liquibaseCommandValue> date/time.
rollbackToDateSql - Generate the SQL to rollback changes made to the database since the <liquibaseCommandValue> date/time.
snapshot - Capture the current state of the database.
snapshotReference
status - Generate a list of pending changesets.
syncHub - Synchronize the local DatabaseChangeLog table with Liquibase Hub.
tag - Mark the current database state with the <liquibaseCommandValue> tag.
tagExists - Verify the existence of the <liquibaseCommandValue> tag.
unexpectedChangesets - Generate a list of changesets that have been executed but are not in the current changelog.
update - Deploy any changes in the changelog file that have not been deployed.
updateCount - Deploy the next <liquibaseCommandValue> changes from the changelog file.
updateCountSql - Generate the SQL to deploy the next <liquibaseCommandValue> changes from the changelog file.
updateOneChangeset - deploy any specific changeset in your changelog to your database. It is only available for Liquibase Pro users
updateOneChangesetSql - Output the SQL to deploy any specific changeset in your changelog to your database. It is only available for Liquibase Pro users
updateSql - Writes SQL to update the database to the current version to <liquibaseOutputFile> or STDOUT of no file is specified.
updateTestingRollback - Updates database, then rolls back changes before updating again. Useful for testing rollback support.
updateToTag - Deploy changes from the changelog file to the <liquibaseCommandValue> tag.
updateToTagSql - Generate the SQL to deploy changes from the changelog file to the <liquibaseCommandValue> tag]
validate - Validate the changelog for errors.

Verification tasks
------------------
check - Runs all checks.
test - Runs the test suite.

Rules
-----
Pattern: clean<TaskName>: Cleans the output files of a task.
Pattern: build<ConfigurationName>: Assembles the artifacts of a configuration.

To see all tasks and more detail, run gradlew tasks --all

To see more detail about a task, run gradlew help --task <task>

BUILD SUCCESSFUL in 615ms
4 actionable tasks: 1 executed, 3 up-to-date
```



### Steps To Reproduce

Install https://github.com/liquibase/liquibase-gradle-plugin with Spring Boot v3 and try to integrate it with Hibernate, PostgreSQL.

### Expected/Desired Behavior

Do not produce exceptions, or at least suggest to me what is wrong / missing.

### Liquibase Version

_No response_

### Database Vendor & Version

PostgreSQL 15

### Liquibase Integration

spring boot, gradle

### Liquibase Extensions

_No response_

### OS and/or Infrastructure Type/Provider

macOS

### Additional Context

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR (Thank you!)

--- a/liquibase-standard/src/main/java/liquibase/executor/jvm/JdbcExecutor.java
+++ b/liquibase-standard/src/main/java/liquibase/executor/jvm/JdbcExecutor.java
@@ -102,7 +102,9 @@ public class JdbcExecutor extends AbstractExecutor {
     }
 
     private void showSqlWarnings(Statement stmtToUse) throws SQLException {
-        if (! SqlConfiguration.SHOW_SQL_WARNING_MESSAGES.getCurrentValue() || stmtToUse.getWarnings() == null) {
+        if (Boolean.TRUE.equals(! SqlConfiguration.SHOW_SQL_WARNING_MESSAGES.getCurrentValue() ||
+            stmtToUse == null) ||
+            stmtToUse.getWarnings() == null) {
             return;
         }
         SQLWarning sqlWarning = stmtToUse.getWarnings();

--- a/liquibase-standard/src/main/java/liquibase/snapshot/jvm/JdbcSnapshotGenerator.java
+++ b/liquibase-standard/src/main/java/liquibase/snapshot/jvm/JdbcSnapshotGenerator.java
@@ -140,11 +140,16 @@ public abstract class JdbcSnapshotGenerator implements SnapshotGenerator {
             } else {
                 catalogs = ((JdbcConnection) database.getConnection()).getMetaData().getCatalogs();
             }
-            while (catalogs.next()) {
-                if (((AbstractJdbcDatabase) database).jdbcCallsCatalogsSchemas()) {
-                    returnList.add(catalogs.getString("TABLE_SCHEM"));
-                } else {
-                    returnList.add(catalogs.getString("TABLE_CAT"));
+            if (catalogs == null) {
+                String message = "Unable to retrieve the list of catalog names from the database metadata";
+                Scope.getCurrentScope().getLog(getClass()).warning(message);
+            } else {
+                while (catalogs.next()) {
+                    if (((AbstractJdbcDatabase) database).jdbcCallsCatalogsSchemas()) {
+                        returnList.add(catalogs.getString("TABLE_SCHEM"));
+                    } else {
+                        returnList.add(catalogs.getString("TABLE_CAT"));
+                    }
                 }
             }
         } finally {
@@ -158,7 +163,7 @@ public abstract class JdbcSnapshotGenerator implements SnapshotGenerator {
             
         }
 
-        if (returnList.size() == 0) {
+        if (returnList.isEmpty()) {
             returnList.add(database.getDefaultCatalogName());
         }
         return returnList.toArray(new String[0]);

